### PR TITLE
fix: fix <MediaSensor> event handling

### PR DESCRIPTION
**Pull request by streamich**

### CODE REVIEW

File: src/MediaSensor/__tests__/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    });
+
+    const MyComp = jest.fn();
+
+    MyComp.mockImplementation(() => null);
+
+    const wrapper = mount(<MediaSensor query='(foobar)' comp={MyComp}></MediaSensor>);
+
+    expect(MyComp).toHaveBeenCalledTimes(1);
+    expect(MyComp.mock.calls[0][0]).toEqual({matches: true});
```
Comment (xram) [inline src/MediaSensor/__tests__/index.test.tsx]: Did you consider using [`toHaveBeenCalledWith`](https://facebook.github.io/jest/docs/en/expect.html#tohavebeencalledwitharg1-arg2-)?

Comment (streamich) [reply src/MediaSensor/__tests__/index.test.tsx]: Yes, but here I want to check only the first argument.

File: src/MediaSensor/__tests__/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    expect(matchMediaReturn.removeListener.mock.calls[0][0]).toBe(listener);
+  });
+
+  describe('when changing media query dynamically', () => {
+    it('re-attaches listener and removes old one', () => {
+      const [matchMedia, matchMediaReturn] = createMatchMedia() as any;
+      let listeners = [];
+
+      matchMediaReturn.addListener.mockImplementation((listener) => {
+        listeners.push(listener);
```
Comment (xram) [inline src/MediaSensor/__tests__/index.test.tsx]: Is there any point in maintaining the `listeners` array? It doesn't appear to be used anywhere.

Comment (streamich) [reply src/MediaSensor/__tests__/index.test.tsx]: Good spot, I've made use of it.

**Pull request merged**